### PR TITLE
Add GoogleFetcher

### PR DESCRIPTION
**Pull request by bartek**
This allows the fetching of items using files.get from Google Drive

### CODE REVIEW

**Review by tballison — CHANGES_REQUESTED** [review]
Do you want to merge this into main? Or actually just the tika-grpc-3x-features?

If we merge to main, I'll cherrypick back to 3x.

We're 4x/main is now the dev branch.

**Comment by bartek** [comment]
@tballison Thanks for review! Honestly, I wasn't expecting one as I'm mostly pushing this to collaborate with @nddipiazza, however, if it makes sense to work as a group, then I will happily do so.

I am pushing to `tika-grpc-3x-features` as that is the branch we're working off in terms of using the tika-pipes project, but I'm not sure on the overall state of things -- if you have a clearer picture, it'd be great to know.

Happy to hop on a quick call if it's easier. I'll read your other comments in the meantime :+1:

**Comment by bartek** [comment]
> To the degree we can make small/logical changes in main to achieve the project's goals, all the better? This is definitely a standalone PR that can go straight into main, I think.

Sounds great, I'm glad you think so. Let me sort out your feedback and then we'll go from there. Appreciate the response!

File: tika-pipes/tika-fetchers/tika-fetcher-google/pom.xml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>tika-fetchers</artifactId>
+        <groupId>org.apache.tika</groupId>
+        <version>3.0.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (bartek) [reply tika-pipes/tika-fetchers/tika-fetcher-google/pom.xml:9]: 3.0.0-SNAPSHOT is referenced in many of the pom.xml files inside tika-pipes. I can update this Google Fetcher to reference `4.0.0-SNAPSHOT`, but should there be a separate PR to get everything else on the same version identifier?

File: tika-pipes/tika-fetchers/tika-fetcher-google/pom.xml [hunk 2]
```diff
@@ -0,0 +1,96 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
[the remark this answers is not in the document]
Comment (bartek) [reply tika-pipes/tika-fetchers/tika-fetcher-google/pom.xml:20]: Added in fixup: https://github.com/apache/tika/pull/2074/commits/40476c19f7fcfdb98fc0e80f892bd13b4ab5a664

File: tika-pipes/tika-fetchers/tika-fetcher-google/src/main/java/org/apache/tika/pipes/fetchers/google/GoogleFetcher.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                InputStream is = driveService.files()
+                        .get(fileId)
+                        .executeMediaAsInputStream();
+
+                if (is == null) {
+                    throw new IOException("Empty input stream when we tried to parse " + fetchKey);
+                }
+
+                if (spoolToTemp) {
+                    File tempFile = Files.createTempFile("spooled-temp", ".dat").toFile();
```
[the remark this answers is not in the document]
Comment (bartek) [reply tika-pipes/tika-fetchers/tika-fetcher-google/src/main/java/org/apache/tika/pipes/fetchers/google/GoogleFetcher.java]: Thanks, used in this fixup: https://github.com/apache/tika/pull/2074/commits/0823d12a60f7dfb6f45e724f556ecb2e791dbcd6

File: tika-pipes/tika-fetchers/tika-fetcher-google/src/main/java/org/apache/tika/pipes/fetchers/google/GoogleDriveFetcher.java
Comment (bartek) [inline tika-pipes/tika-fetchers/tika-fetcher-google/src/main/java/org/apache/tika/pipes/fetchers/google/GoogleDriveFetcher.java:1]: Renamed to GoogleDriveFetcher as the [files.get](https://developers.google.com/drive/api/reference/rest/v3/files/get) call is part of the Google Drive API

I don't expect this fetcher to service other Google APIs, and the fetchKey/configuration is kept simple by narrowing the fetchers scope

File: tika-pipes/tika-fetchers/tika-fetcher-google/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bartek) [reply tika-pipes/tika-fetchers/tika-fetcher-google/pom.xml]: I pushed https://github.com/apache/tika/pull/2074/commits/da30edef4bf7f62468cd1a21abdb0894e48e61a5, hopefully this satisfies what you were expecting?

I can also adjust the merge-to branch to be `main`. Are all the changes inside tika-grpc-3x-features in main? (excluding this one)

**Comment by tballison** [comment]
Sorry, what I meant was that over on the tika-grpc-3x branch you should change everything to 4.0.0-SNAPSHOT. On this PR, you should only need to do that for the GoogleDriveFetcher, and you should make this PR against `main`.

**Comment by bartek** [comment]
> Sorry, what I meant was that over on the tika-grpc-3x branch you should change everything to 4.0.0-SNAPSHOT. On this PR, you should only need to do that for the GoogleDriveFetcher, and you should make this PR against main.

No worries, made that change here: https://github.com/apache/tika/pull/2074/commits/d439aca4bef3960215b7c363b00886a318fb3937

**Comment by bartek** [comment]
Closing this in favour of https://github.com/apache/tika/pull/2077, which is based off `main` and targeting it. The tika-grpc-3x-features branch needs significant cleanup.

**Pull request closed without merging**